protocol/x/vault/keeper/params_test.go

```diff
@@ -0,0 +1,45 @@
+package keeper_test
+
+import (
+	"testing"
+
+	testapp "github.com/dydxprotocol/v4-chain/protocol/testutil/app"
+	"github.com/dydxprotocol/v4-chain/protocol/x/vault/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetSetParams(t *testing.T) {
+	tApp := testapp.NewTestAppBuilder(t).Build()
+	ctx := tApp.InitChain()
+	k := tApp.App.VaultKeeper
+
+	// Params should have default values at genesis.
+	params := k.GetParams(ctx)
+	require.Equal(t, types.DefaultParams(), params)
+
+	// Set new params and get.
+	newParams := types.Params{
+		Layers:                 3,
+		SpreadMinPpm:           4_000,
+		SpreadBufferPpm:        2_000,
+		SkewFactorPpm:          999_999,
+		OrderSizePpm:           200_000,
+		OrderExpirationSeconds: 10,
+	}
+	err := k.SetParams(ctx, newParams)
+	require.NoError(t, err)
+	require.Equal(t, newParams, k.GetParams(ctx))
+
+	// Set invalid params and get.
+	invalidParams := types.Params{
+		Layers:                 3,
+		SpreadMinPpm:           4_000,
+		SpreadBufferPpm:        2_000,
+		SkewFactorPpm:          1_000_000, // invalid
```

Comment:
talked about this in slack, just repeating here for clarity. I think we should allow skew greater than 1. I think the theoretical bound should be 1/order_size_pct, but don't know if we actively need to check this for v0.

Reply:
will remove check